### DISCUSSION TITLE
scale conversion from surface.pix to vehicle coordinates

### DISCUSSION THREAD

**Discussion by rohanchandra30** [eleurent/highway-env#174] [General]
thanks for this wonderful work!

Suppose I wanted to draw a line connecting 2 vehicles in the straight road environment. I can get the distance between the vehicles' coordinates. Would that correspond equivalently to the pixel distance between the 2 vehicles if I decide to draw a line between them?

Or is there a different scale conversion from vehicle coordinate to surface.pix?

**Comment by eleurent** [comment]
Hi @rohanchandra30, thanks!

Conversions between euclidean state space and pixel space are performed in the [WorldSurface](https://github.com/eleurent/highway-env/blob/master/highway_env/road/graphics.py#L17) class.

* [pix(self, length: float) -> int](https://github.com/eleurent/highway-env/blob/master/highway_env/road/graphics.py#L37) converts a length (in meters) to pixels, which I think is what you asked for.
* [pos2pix(self, x: float, y: float) -> Tuple[int, int]:](https://github.com/eleurent/highway-env/blob/master/highway_env/road/graphics.py#L46) converts a world position (x,y) to a screen coordinate in pixels.
* [vec2pix(self, vec: PositionType) -> Tuple[int, int]:](https://github.com/eleurent/highway-env/blob/master/highway_env/road/graphics.py#L56) is similar but with an array-like input position.

Thus, if you wish to draw a line between two vehicles, you can directly use
`pygame.draw.line(surface, color, surface.pos2pix(vehicle1.position), surface.pos2pix(vehicle2.position), width)`
